services/frontend-service/src/ui/utils/AzureAuthProvider.tsx

```diff
@@ -0,0 +1,123 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+import * as React from 'react';
+import { BrowserHeaders } from 'browser-headers';
+import { Configuration, PublicClientApplication } from '@azure/msal-browser';
+import {
+    MsalProvider,
+    AuthenticatedTemplate,
+    useIsAuthenticated,
+    UnauthenticatedTemplate,
+    useMsal,
+} from '@azure/msal-react';
+import { GetFrontendConfigResponse } from '../../api/api';
+import { createStore } from 'react-use-sub';
+import { grpc } from '@improbable-eng/grpc-web';
+
+export const [useFrontendConfig, UpdateFrontendConfig] = createStore({} as GetFrontendConfigResponse);
+export const [useReady, UpdateReady] = createStore({ config: 'waiting', auth: 'waiting' });
```

Comment:
can we use an enum here instead of just strings? (for both `config` and `waiting`)

Reply:
I'd say probably yes. I was actually interested in getting feedback about this. Does this `Ready` thing make any sense? is it best practices? any better way to do it? 

the goal of this is to only subscribe to the overview when we got everything ready.

my other idea is to add a flag to the other store, e.g. (useFrontendConfig)?

Reply:
> the goal of this is to only subscribe to the overview when we got everything ready.

is there a state where only half of it is ready? If yes, it might make sense to have a flag/state like this. 
But what is this in-between state? I'm not sure if I get that.